Add Docs For Prometheus Metrics

seanmalloy opened the pull request:
Prometheus metrics were added as part of the Vault v1.1.0 release in PR #5308. But no documentation was created. Adds the telemetry configuration docs and the API docs.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/vault?pullRequest=6434)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/vault?pullRequest=6434)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/vault?pullRequest=6434).</sub>

riuvshin commented:
It seems that API Docs also should be updated: https://www.vaultproject.io/api/system/

kalafut commented:
@seanmalloy re: the OpenAPI output, query parameters in general aren't being handled correctly. I'll look into fixing that.

seanmalloy commented:
I added the missing API docs for /v1/sys/metrics. Let  me know if any changes are needed.

kalafut commented on website/source/api/system/metrics.html.md:
```diff
@@ -0,0 +1,69 @@
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `GET`    | `/sys/metrics`               | `000 application/json` |
```
The "Produces" column can be removed. We just today made this change broadly (https://github.com/hashicorp/vault/commit/d096f62d8dd49836bf0500bd33942f7dde85f5df).

seanmalloy replied:
Fixed

kalafut commented on website/source/docs/configuration/telemetry.html.md:
```diff
@@ -135,3 +135,31 @@ These `telemetry` parameters apply to
+  prometheus metrics are retained in memory. It is recommended to also enable the option
+  `disable_hostname` to avoid having prefixed metrics with hostname. Vault does not use the
+  default Prometheus path, so Prometheus must be configured as follows. Note that using
+  ?format=prometheus in the path won't work as ? will be escaped, so it must be specified
```
```suggestion
  `?format=prometheus` in the path won't work as "?" will be escaped, so it must be specified
```

kalafut commented on website/source/docs/configuration/telemetry.html.md:
```diff
@@ -135,3 +135,31 @@ These `telemetry` parameters apply to
+  `disable_hostname` to avoid having prefixed metrics with hostname. Vault does not use the
+  default Prometheus path, so Prometheus must be configured as follows. Note that using
+  ?format=prometheus in the path won't work as ? will be escaped, so it must be specified
+  as a parameter. Addtionally a Vault token is required to access /v1/sys/metrics. The prometheus
```
```suggestion
  as a parameter. Additionally a Vault token is required to access /v1/sys/metrics. The prometheus
```

kalafut commented:
This looks good! Thanks for putting it together. I left a few inline comments. In addition, since the metrics page is new, there is another change needed in order for it to show up in the sidebar. Please add `metrics` before `mounts` here:  https://github.com/hashicorp/vault/blob/3af9be469150b76aa035545f522a7efbf36ffc6d/website/source/layouts/api.erb#L117

I'll handle the OpenAPI updates separately.

seanmalloy commented:
@kalafut thanks for the review. I have addressed all of your review comments. Let me know if you want me to clean up the commit history before merging.

kalafut commented:
@seanmalloy Looks good, thanks again!

kalafut commented:
OpenAPI updates:  https://github.com/hashicorp/vault/pull/6490
